From 0c3177e0530ef3d97fe7bd912ace8ec13cc94047 Mon Sep 17 00:00:00 2001
From: Sebastian Helzle <sebastian@helzle.it>
Date: Thu, 12 Jun 2025 13:15:33 +0200
Subject: [PATCH] BUGFIX: Prevent sharing splitString between node routings

Resolves: #5571
---
 .../EventSourcedFrontendNodeRoutePartHandler.php                | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php b/Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php
index 897fad3c9e..b2461e22fd 100644
--- a/Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php
+++ b/Neos.Neos/Classes/FrontendRouting/EventSourcedFrontendNodeRoutePartHandler.php
@@ -141,8 +141,6 @@
  * if needed.
  *
  * Then, the {@see DimensionResolverInterface} of the target site is called for adjusting the URL.
- *
- * @Flow\Scope("singleton")
  */
 final class EventSourcedFrontendNodeRoutePartHandler extends AbstractRoutePart implements
     DynamicRoutePartInterface,
